src/commands/src/shared/next-gen-parse.ts

```diff
@@ -1,33 +1,45 @@
+import path from 'path';
+import { Job } from '../../../entities/job';
+import { getDirectory } from '../../../job/jobHandler';
 import { CliCommandResponse, executeCliCommand } from '../helpers';
 
 const RSTSPEC_FLAG = '--rstspec=https://raw.githubusercontent.com/mongodb/snooty-parser/latest/snooty/rstspec.toml';
 interface NextGenParseParams {
-  repoDir: string;
-  commitHash: string;
-  patchId?: string;
+  job: Job;
   isProd?: boolean;
 }
-export async function nextGenParse({
-  repoDir,
-  patchId,
-  commitHash,
-  isProd,
-}: NextGenParseParams): Promise<CliCommandResponse> {
-  const commandArgs = ['build', repoDir, '--output', `${repoDir}/bundle.zip`, RSTSPEC_FLAG];
+export async function nextGenParse({ job, isProd }: NextGenParseParams): Promise<CliCommandResponse> {
```

Comment:
I was looking at it from the perspective of each individual function more. For example, `nextGenParse` only uses `job.payload.newHead` and `job.payload.patch`, but `job` has many more properties than that. If we were to test this function, mocking out a fake `job` object would be a bit tedious.

With that said, I think you have a valid point that the `jobHandler` would then be much larger having to break out the individual properties that we want to pass to each function, which definitely makes it less readable from that perspective. And like you mentioned, some of the other functions have many more properties that are used from the `job` object. 

To be honest, I am totally fine keeping it as is since either way there are some trade-offs from which perspective you are looking at. Thanks for the thoughtful response!